src/js/plugins/search/view.js

```diff
@@ -169,22 +166,48 @@ class SearchView {
   showPrevResult() {
     if (this.currentMatchIndex === 0) { return; }
     if (this.br.mode === this.br.constModeThumb) { this.br.switchMode(this.br.constMode1up); }
-    if (!~this.currentMatchIndex) { this.currentMatchIndex = 1; }
+    if (!~this.currentMatchIndex) {
+      this.currentMatchIndex = this.getClosestMatchIndex((start, end, comparator) => end[0] > comparator) + 1;
+    }
     this.br.$('.BRnavline .BRsearch').eq(--this.currentMatchIndex).click();
     this.updateResultsPosition();
+    this.updateSearchNavigationButtons();
   }
 
   showNextResult() {
     if (this.currentMatchIndex + 1 === this.matches.length) { return; }
     if (this.br.mode === this.br.constModeThumb) { this.br.switchMode(this.br.constMode1up); }
+    if (!~this.currentMatchIndex) {
+      this.currentMatchIndex = this.getClosestMatchIndex((start, end, comparator) => start[start.length - 1] > comparator) - 1;
+    }
     this.br.$('.BRnavline .BRsearch').eq(++this.currentMatchIndex).click();
     this.updateResultsPosition();
+    this.updateSearchNavigationButtons();
+  }
+
+  getClosestMatchIndex(comparisonFn) {
+    const matchPages = this.matches.map((m) => m.par[0].page);
+    const currentPage = this.br.currentIndex() + 1;
+    const closestTo = (pool, comparator) => {
```

Comment:
this is cool! can we add a mini doc on its goodness?

Reply:
You mean explaining the binary search algorithm? I could add something in there.